### DISCUSSION TITLE
Versioning scheme

### DISCUSSION THREAD

**Discussion by iron3oxide** [yezz123/ormdantic#22] [General]
What is the idea behind the versioning scheme of this project? From what I could tell, it goes as follows:

- starts at 1.0.0
- every significant PR triggers a new release
- new releases mean the second digit gets increased by one, e.g. 1.1.0

It seems to be standard for projects that are not yet ready for production to start version at something like 0.0.1, but this can not be easily changed anymore, so this gripe of mine can be disregarded.

I wonder though what would trigger a release with an increase in the third digit, e.g. 1.2.1? 
A bug fix?

**Comment by yezz123** [comment]
> What is the idea behind the versioning scheme of this project? From what I could tell, it goes as follows:
> 
> * starts at 1.0.0
> * every significant PR triggers a new release
> * new releases mean the second digit gets increased by one, e.g. 1.1.0
> 
> It seems to be standard for projects that are not yet ready for production to start version at something like 0.0.1, but this can not be easily changed anymore, so this gripe of mine can be disregarded.
> 
> I wonder though what would trigger a release with an increase in the third digit, e.g. 1.2.1? A bug fix?

In That Case, the first release was the major one I mean `1.0.0`, and the changes from `1.X.0` mean the changes that will not break the use case of the package, like this one for example #18 , it's just a fix for the connection string so it will be just a minor change, that's why I version the package in that way!

Also, another case is that each dependency used by Ormdantic is used in production from the first version which means no problem starting from 1.0.0 and then Updating it based on the amount of change ex (breaking the codebase, minors, or fixing bugs).

**Reply by iron3oxide** [reply]
I see, but if something is just a minor change compared to 1.2.0, why not call the release 1.2.1 instead of 1.3.0? That way one would actually use the last digit and there would be a clear distinction between three cases:

1. Breaking/major update -> X.0.0
2. Significant non-breaking update (big new feature/important fix for example) -> 1.X.0
3. Minor update/fix -> 1.0.X

EDIT: I just noticed that the last sentence of your reply probably meant what I proposed here. If #18 is a fix though, my question stands: when is the last digit of the version number used?

**Reply by yezz123** [reply]
Well, the last digit is used for mini changes and just some kind of small bugs, but the change I made is somehow a medium change that needs a Significant non-breaking update, that's why!
